packages/flutter_test/lib/src/nonconst.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// This function can be used to prevent some const lints.
+///
+/// It's usually used when we want several instances to be separate instances so
+/// that we're not just checking with a single object.
+///
+/// ```dart
+/// class A {
+///   const A(this.i);
+///   int i;
+/// }
+///
+/// main () {
+///   // prevent prefer_const_constructors lint
+///   new A(nonconst(null));
+///
+///   // prevent prefer_const_declarations lint
+///   final int $null = nonconst(null);
+///   final A a = nonconst(const A(null));
+/// }
+/// ```
+T nonconst<T>(T t) => t;
```

Comment:
This is all great.

I think we should make a slight change to the framing in the documentation for nonconst. Rather than saying it's to avoid lints, I think we should say it's to convert a const literal into a non-const value, so that it can be used to call a const constructor in such a way as to create a new instance rather than creating the common const instance.

Also, we shouldn't use "we" or "you" in the docs; our style is to use the passive voice.